src/main/java/game_world/validators/EventInputValidator.java

```diff
@@ -0,0 +1,39 @@
+package game_world.validators;
+
+import game_world.objects.Action;
+import io.InputValidator;
+
+import java.util.ArrayList;
+
+
+public class EventInputValidator implements InputValidator {
+
+    private final Action currentAction;
+    private final ArrayList<String> possibleInputs;
+
+    public EventInputValidator(Action currentAction) {
+        this.currentAction = currentAction;
+        this.possibleInputs = new ArrayList<>();
+    }
+
+    public EventInputValidator(Action currentAction, ArrayList<String> possibleInputs) {
+        this.currentAction = currentAction;
+        this.possibleInputs = possibleInputs;
+    }
+
+    @Override
+    public String parseAndValidate(String input) {
+        // Validates and parses user input if valid, returns null if invalid.
+        if (currentAction == Action.ARBITRARY) {
```

Comment:
this seems to be the only place action is used, and can easily be changed by for example setting possibleInputs to null, and if possibleInputs is null then it just returns input as-is. 
Is action really needed in this class?